module/block_iterator/height_based/iterator.go

```diff
@@ -0,0 +1,76 @@
+package height_based
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/storage"
+)
+
+type HeightIterator struct {
+	// dependencies
+	headers  storage.Headers
```

Comment:
only the `BlockIDByHeight` function is needed from headers. Consider just using a `func (height) flow.Identifier`

Reply:
That's a good idea. Actually if I change it into a `GetBlockIDByHeight` function, the whole HeightIterator will be almost identical to ViewIterator, meaning we could just use HeightIterator as ViewIterator by passing a GetBlockIDByView function as GetBlockIDByHeight.

Reply:
if you go this route, would we still need all of the interfaces? one concern I have is all the interfaces add complexity that reader has to wrap their head around to understand the very simple implementation. 

If this could be condensed into a single implementation with a provider function, I think most of that complexity could be removed.

Reply:
Yeah, make sense. I will keep these interfaces for now, and maybe refactor them [in the next PR](https://github.com/onflow/flow-go/pull/6890) after this is merged.